### PR TITLE
feat: pulsar oauth

### PR DESCRIPTION
**Pull request by tabVersion**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

* use pulsar API for enumerator
* support pulsar oauth

integration with stream native cloud

## Checklist For Contributors

- [x] I have written necessary rustdoc comments
~~- [ ] I have added necessary unit tests and integration tests~~
~~- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).~~
- [x] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Checklist For Reviewers

- [x] I have requested macro/micro-benchmarks as this PR can affect performance substantially, and the results are shown.

## Documentation

- [ ] My PR **DOES NOT** contain user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Connector (sources & sinks)

### Release note

remove field: `admin.url`
new fields: `oauth.issuer.url`, `oauth.credentials.url`, `oauth.audience` and `oauth.scope`

|field|required|discription|
|--|--|--|
|`topic`|yes|if `topic` starts with `non-persistent://`, it can never seek to a specific position and will always read from latest with no exactly once guarantee.|
|`service.url`|yes|service URL for pulsar, starts with `pulsar://` or `pulsar+ssl://`|
|`scan.startup.mode`|no|default: `earliest`. accept one of `earliest`, `latest`|
|`scan.startup.timestamp_millis`|no|if specified, the reader will start reading from the timestamp. note that if the topic is `non-persistent`, it will read from the latest. |
|`auth.token`|no|a token for auth|
|`oauth.issuer.url`|conditional|the issuer URL for OAuth2|
|`oauth.credentials.url`|conditional|the path for credential files, starts with `file://`|
|`oauth.audience`|conditional|audience for OAuth2|
|`oauth.scope`|conditional|scope for OAuth2|

Note that if user both set`oauth.*` auth and `auth.token`, only `oauth` authorisation is effective. 

if `oauth.*` provided 

|field|required|
|--|--|
|`oauth.issuer.url`|yes|
|`oauth.credentials.url`|yes|
|`oauth.audience`|yes|
|`oauth.scope`|no|

we require admin privileges for the role. 

</details>

### CODE REVIEW

**Comment by tabVersion** [comment]
it involves auth part, manually tested on stream native cloud

File: src/connector/src/source/pulsar/mod.rs [hunk 1]
```diff
@@ -43,4 +60,36 @@ pub struct PulsarProperties {
...
+            let auth_params = OAuth2Params {
+                issuer_url: oauth.issuer_url.clone(),
+                credentials_url: oauth.credentials_url.clone(),
+                audience: Some(oauth.audience.clone()),
+                scope: oauth.scope.clone(),
+            };
+
+            pulsar_builder = pulsar_builder
+                .with_auth_provider(OAuth2Authentication::client_credentials(auth_params));
+        } else if let Some(auth_token) = &self.auth_token {
```
Comment (lmatz) [inline src/connector/src/source/pulsar/mod.rs:86]: It seems `auth_token` and `oauth` are mutually exclusive?

Do we need to inform users that although both are set, only `auth_token` will be considered?

**Review by lmatz — APPROVED** [review]
LGTM

File: src/connector/src/source/pulsar/mod.rs (hunk 1 above)
Comment (tabVersion) [reply src/connector/src/source/pulsar/mod.rs:86]: pulsar supports these two auth methods. I think in most cases, user won't provide both fields. and of course, I will mention it in doc

**Pull request merged**